### PR TITLE
Storage Stats thing

### PR DESCRIPTION
**Pull request by modakshantanu**

### CODE REVIEW

File: src/main/java/com/algosenpai/app/stats/UserStats.java [hunk 1]
```diff
@@ -32,9 +28,8 @@
 public class UserStats {
     private String userName;
     private String gender;
-    private String level;
-    private String expLevel;
-    private String userDataFilePath;
+    private int level;
```
Comment (modakshantanu) [inline src/main/java/com/algosenpai/app/stats/UserStats.java]: Change them to ints from String

File: src/main/java/com/algosenpai/app/stats/UserStats.java [hunk 2]
```diff
@@ -45,40 +40,10 @@
     //Maps the chapter names to an index value
     private HashMap<String, Integer> chapterNumber;
 
-    /**
```
Comment (modakshantanu) [inline src/main/java/com/algosenpai/app/stats/UserStats.java]: Delete the constructor

File: src/main/java/com/algosenpai/app/stats/UserStats.java [hunk 3]
```diff
@@ -116,15 +81,13 @@ public ChapterStat getStatsByIndex(int index) {
...
-     * @param wasAnsweredCorrectly Whether the user answered correctly or incorrectly.
+     * TODO.
      */
-    public void updateCurrentChapter(boolean wasAnsweredCorrectly) {
-        currentChapter.totalAnswered++;
-        if (wasAnsweredCorrectly) {
-            currentChapter.correctAnswers++;
-        }
-        currentChapter.recalculateStats();
+    public void updateChapter(int index, int totalAnswered, int correct) {
```
Comment (modakshantanu) [inline src/main/java/com/algosenpai/app/stats/UserStats.java]: We r not using currentChapter anymore,  directly update the chapterData with a new batch of correctly and incorrect questions

File: src/main/java/com/algosenpai/app/stats/UserStats.java [hunk 4]
```diff
@@ -293,13 +256,13 @@ public static UserStats getDefaultUserStats() {
         chapters.add(new ChapterStat("Sorting",1,0,0,0,0,0,""));
         chapters.add(new ChapterStat("Linked List",2,0,0,0,0,0,""));
         chapters.add(new ChapterStat("Bitmask",3,0,0,0,0,0,""));
-        return new UserStats("DefaultName", "male", "1", "0", chapters);
+        return new UserStats("DefaultName", "male", 1, 0, chapters);
     }
```
Comment (modakshantanu) [inline src/main/java/com/algosenpai/app/stats/UserStats.java]: edit parseString to parse the new format (with 2 new parameters)

File: src/main/java/com/algosenpai/app/storage/Storage.java [hunk 5]
```diff
@@ -32,24 +35,31 @@ public static String loadData(String filePath) throws IOException {
...
+            fw.write(data);
+            fw.close();
+        } catch (IOException e) {
+            e.printStackTrace();
         }
-
-        FileWriter fw = new FileWriter(filePath);
-        fw.write(data);
-        fw.close();
     }
```
Comment (modakshantanu) [inline src/main/java/com/algosenpai/app/storage/Storage.java]: Rewriting to remove the exceptions

File: src/main/java/com/algosenpai/app/ui/Ui.java [hunk 6]
```diff
@@ -89,10 +90,21 @@ public Ui() throws FileNotFoundException {
...
+        } else {
+            dialogContainer.getChildren().add(DialogBox.getSenpaiDialog(
+                    "Welcome back " + tempStats.getUsername(), senpaiImage));
+            userPic.setImage(userImage);
+            levelProgress.setProgress(0);
+            playerLevel.setText("You are Level " + tempStats.getUserLevel());
+        }
+
+
         handle();
```
Comment (modakshantanu) [inline src/main/java/com/algosenpai/app/ui/Ui.java]: on initialize, display the`GREETING_MESSAGE` only if the stats name is "DefaultName" , indicating that the user has launched the program for the first time

Else skip that and just say welcome + name

File: src/test/java/com/algosenpai/app/StorageTest.java [hunk 7]
```diff
@@ -1,135 +1,135 @@
...
+                    random.nextInt(),
+                    random.nextInt(),
+                    random.nextDouble() * random.nextInt(),
+                    RandomString.make(random.nextInt(100) + 1)
+            );
+            chapterStats.add(chapterStat);
+        }
+        return new UserStats("Test Test Test", "Test test Test", 0,0,chapterStats);
+    }
+}
```
Comment (modakshantanu) [inline src/test/java/com/algosenpai/app/StorageTest.java]: Put back test

File: src/test/java/com/algosenpai/app/UserStatsTest.java [hunk 8]
```diff
@@ -1,56 +1,56 @@
...
+    /**
+     * Tests if parsing works correctly if all the parameters of userStats are blank.
+     */
+    @Test
+    public void stringParsing_emptyStats() {
+        UserStats userStats = new UserStats("","",0,0,new ArrayList<>());
+        UserStats copy = UserStats.parseString(userStats.toString());
+        assertEquals(copy,userStats);
+    }
+}
```
Comment (modakshantanu) [inline src/test/java/com/algosenpai/app/UserStatsTest.java]: Put back test

File: src/main/java/com/algosenpai/app/MainApp.java [hunk 9]
```diff
@@ -44,16 +44,14 @@
         }
     }
 
-    public MainApp() throws IOException {
```
[the remark this answers is not in the document]
Comment (sevenmatt7) [reply src/main/java/com/algosenpai/app/MainApp.java]: nice

**Pull request merged**